js/menu/filter/filter.js

```diff
@@ -25,54 +34,145 @@ filterButton.addEventListener("click", () => {
   }
 });
 
-const filters = document.getElementById("filters");
-const apply = document.getElementById("filter-apply");
-const reset = document.getElementById("filter-reset");
-
-let parametersRange = ["momentum", "vertex", "time", "mass", "charge"];
+export function renderRangeParameters(container, rangeParameters) {
+  const rangeFilters = document.createElement("div");
+  rangeFilters.id = "range-filters";
+  rangeFilters.style.display = "grid";
+  rangeFilters.style.width = "fit-content";
+  rangeFilters.style.columnGap = "10px";
+  rangeFilters.style.rowGap = "5px";
+  rangeFilters.style.alignItems = "center";
+  rangeFilters.style.gridTemplateColumns =
+    "fit-content(100%) fit-content(100%)";
+  rangeParameters.forEach((parameter) => {
+    parameter.min = undefined;
+    parameter.max = undefined;
+    parameter.render(rangeFilters);
+  });
+  container.appendChild(rangeFilters);
+}
+
+export function getWidthFilterContent() {
+  const filterContent = document.getElementById("filter-content");
+  filterContent.style.display = "flex";
+  const rangeFilters = document.getElementById("range-filters");
+  const width = rangeFilters.offsetWidth;
+  filterContent.style.display = "none";
+  return `${width}px`;
+}
+
+export function renderGenSim(sim, gen, container) {
+  const div = document.createElement("div");
+  div.style.display = "grid";
+  div.style.width = "fit-content";
+  div.style.columnGap = "10px";
+  div.style.rowGap = "5px";
+  div.style.alignItems = "start";
+  div.style.gridTemplateColumns = "fit-content(100%) fit-content(100%)";
+  sim.render(div);
+  gen.render(div);
+  container.appendChild(div);
+}
+
+let parametersRange = [
+  {
+    property: "momentum",
+    unit: "GeV",
+  },
+  {
+    property: "mass",
+    unit: "GeV",
+  },
+  {
+    property: "charge",
+    unit: "e",
+  },
+  {
+    property: "vertex",
+    unit: "mm",
+  },
+  {
+    property: "time",
+    unit: "ns",
+  },
+];
+
+parametersRange = parametersRange.sort((a, b) =>
+  a.property.localeCompare(b.property)
+);
 
 parametersRange = parametersRange.map((parameter) => new Range(parameter));
 
-parametersRange.forEach((parameter) => parameter.render(filters));
-
-let bitsCheckbox = [23, 24, 25, 26, 27, 28, 29, 30];
-
-bitsCheckbox = bitsCheckbox.map((bit) => new Checkbox("simStatus", bit));
-
-bitsCheckbox.forEach((checkbox) => checkbox.render(filters));
-
-apply.addEventListener("click", () => {
+const bitFieldDisplayValues = {
```

Comment:
```suggestion
const SimStatusBitFieldDisplayValues = {
```

Just so that it is immediately clear which bit field display values we are talking about here.